src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java

```diff
@@ -1,72 +1,99 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
 import org.springframework.web.client.HttpClientErrorException;
 import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.ccd.AppInsights;
 import uk.gov.hmcts.ccd.ApplicationParams;
 import uk.gov.hmcts.ccd.data.SecurityUtils;
-import uk.gov.hmcts.ccd.domain.model.draft.CreateCaseDraft;
-import uk.gov.hmcts.ccd.domain.model.draft.Draft;
-import uk.gov.hmcts.ccd.domain.model.draft.UpdateCaseDraft;
+import uk.gov.hmcts.ccd.domain.model.draft.*;
 import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
 import uk.gov.hmcts.ccd.endpoint.exceptions.ServiceException;
 
 import javax.inject.Inject;
-import javax.inject.Named;
-import javax.inject.Singleton;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
 
+import static uk.gov.hmcts.ccd.AppInsights.DOC_MANAGEMENT;
+import static uk.gov.hmcts.ccd.AppInsights.DRAFT_STORE;
+import static uk.gov.hmcts.ccd.domain.model.draft.DraftResponseBuilder.aDraftResponse;
 
-@Named
+
+@Service
 @Qualifier(DefaultDraftGateway.QUALIFIER)
-@Singleton
 public class DefaultDraftGateway implements DraftGateway {
 
-    private static final Logger LOG = LoggerFactory.getLogger(DefaultDraftGateway.class);
     public static final String QUALIFIER = "default";
+    public static final String DRAFT_ACCESS_EXCEPTION_MSG = "There is a problem with retrieving drafts.";
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultDraftGateway.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
     private static final String DRAFT_ENCRYPTION_KEY_HEADER = "Secret";
     private static final int RESOURCE_NOT_FOUND = 404;
 
+    @Qualifier("draftsRestTemplate")
+    @Autowired
     private final RestTemplate restTemplate;
     private final SecurityUtils securityUtils;
     private final ApplicationParams applicationParams;
+    private final AppInsights appInsights;
 
     @Inject
     public DefaultDraftGateway(
-            final RestTemplate restTemplate,
-            final SecurityUtils securityUtils,
-            final ApplicationParams applicationParams) {
+        final RestTemplate restTemplate,
+        final SecurityUtils securityUtils,
+        final ApplicationParams applicationParams,
+        final AppInsights appInsights) {
         this.restTemplate = restTemplate;
         this.securityUtils = securityUtils;
         this.applicationParams = applicationParams;
+        this.appInsights = appInsights;
     }
 
     @Override
-    public Draft save(final CreateCaseDraft draft) {
+    public Long save(final CreateCaseDraft draft) {
```

Comment:
not to confuse CreateCaseDraft with our data store operation classes, 
can we rename this to CreateCaseDraftRequest?